src/vs/workbench/services/notification/common/notificationService.ts

```diff
@@ -52,12 +57,63 @@ export class NotificationService extends Disposable implements INotificationServ
 	}
 
 	notify(notification: INotification): INotificationHandle {
-		return this.model.addNotification(notification);
+
+		let handle: INotificationHandle;
+		if (notification.neverShowOptions) {
+			const id = notification.neverShowOptions.id;
+			if (this.isRejected(id)) {
+				return new NoOpNotification();
+			}
+
+			const neverShowAction = new Action(
+				'workbench.dialog.choice.neverShowAgain',
+				nls.localize('neverShowAgain', "Don't Show Again"),
+				undefined, true, () => {
+					handle.close();
+					this.neverShow(id);
+					return Promise.resolve();
+				});
+
+			//default to secondary
```

Comment:
The default cannot be secondary if the option is called isSecondary and is optional. Also it would change the way we used to do it for the large fie optimization case.

Reply:
How about: 
appearance: 'primary' | 'secondary';
?

Reply:
@YisraelV I think the name `isSecondary?: boolean;` is fine, we use it for `IPromptChoice` as well but if not provided, it needs to default to being primary.

Reply:
@bpasero you suggested before the default should be "secondary" in the notify method ([here](https://github.com/microsoft/vscode/pull/73968#issuecomment-498664661)) or did I misunderstand you?

Reply:
@YisraelV I would make it consistent with the prompt choice, not sure what I thought back then.